docs/proposals/14-reversed-cluster-vpn.md

```diff
@@ -0,0 +1,278 @@
+
+A service `vpn-seed-server` of type `ClusterIP` is created for each control plane in its namespace.
+
+The `vpn-shoot-client` pod connects to the correct `vpn-seed-server` service via the SNI passthrough proxy introduced with [SNI Passthrough proxy for kube-apiservers](https://github.com/gardener/gardener/blob/master/docs/proposals/08-shoot-apiserver-via-sni.md) on port 8132. 
+
+Shoot OpenVPN clients (`vpn-shoot-client`) connect to the correct OpenVPN Server using the http proxy feature provided by OpenVPN. A configuration is added to the envoy proxy to detect http proxy requests and open a connection attempt to the correct OpenVPN server.
+
+The `kube-apiserver` to shoot cluster connections are established using the API server proxy feature via an envoy proxy sidecar container of the `vpn-seed-server` container. 
+
+The restriction regarding the `192.168.123.0/24` network range in the current VPN solution still applies to this proposal. No other restrictions are introduced. In the context of this GEP a pull requst has been filed to block usage of that range by shoot clusters.
+
+### Performance and Scalability
+
+We do expect performance and throughput to be slightly lower compared to the existing solution. This is because the OpenVPN server acts as an additional hop and must decrypt and re-encrypt traffic that passes through. As there are no low latency, or high thoughput requirements for this connection we do not assume this to be an issue.
+
+### Availablity and Failure Scenarios
+
+This solution re-uses multiple instances of the envoy component used for the kube-apiserver endpoints. We assume that the availability for kube-apiservers is good enough for the cluster VPN as well. 
+
+The OpenVPN client- and server pods are singleton pods in this approach and therefore are affected by potential failures and during cluster-, and control plane updates. Potential outages are only restricted to single shoot clusters and are comparable to the situation with the existing solution today.
+
+### Feature Gates and Migration Strategy
+
+We have introduced a gardenlet feature gate `ReversedVPN`. If `ManagedIstio`, `APIServerSNI`, and `ReversedVPN` are enabled the proposed solution is automatically enabled for all shoot clusters hosted by the seed. If `ReversedVPN` is enabled but either `ManagedIstio` or `APIServerSNI` is not the gardenlet will panic during startup as this is an invalid configuration. All existing shoot clusters will automatically be migrated during the next reconciliation. We have also added a shoot annotation `alpha.featuregates.shoot.gardener.cloud/reversed-vpn` which can override the feature gate to enable or disable the solution for individual clusters.
```

Comment:
> If `ReversedVPN` is enabled but either `ManagedIstio` or `APIServerSNI` is not the gardenlet will panic

The check for `ManagedIstio` should be removed because it could also be that operators install their own Istio version into the cluster (this is also how `APIServerSNI` works).

Reply:
I am confused, maybe @ScheererJ or @DockToFuture can shed some light on this.

Reply:
Background: If an operator disables `APIServerSNI` there might still be shoots with this annotation, and the next reconciliation could fail in this case (I remember @vpnachev talking about potential panics in this case), so maybe just ignoring the annotation is best?

Reply:
As we need to add an envoy filter to istio (to support http proxy/connect protocol) and that is done as part of the managed istio, I doubt that `ReversedVPN` works out of the box with istio managed by an operator. This is why we added the dependency to `ManagedIstio`.
If you think it should also work without `ManagedIstio`, we should probably document this envoy filter or the requirement that istio needs to support http proxy/connect protocol.
Concerning the inconsistent feature flags, we already added the note above that gardenlet will panic on startup.

Reply:
Aha, interesting, then you might want to add this information to the GEP, please?

Also, cc @mvladev, given that we hopefully strive to make this new VPN 2.0 the default eventually, it means that we can no longer support custom-installed Istio deployments but only our own installation. This might be as well something that should be added to the GEP.

Reply:
Added to the GEP after discussion with @mvladev: ReversedVPN will work with operator managed Istio.